README.mkd

```diff
@@ -54,14 +54,53 @@ base and extensions, e.g. `-march=rv64g`. A target `-march` which includes
 floating point instructions implies a hardfloat calling convention, but can be 
 overridden using the `-mabi` flag (see the next section).
 
-The ISA subset naming conventions are described in Chapter 22 of the RISC-V 
-user-level ISA specification. However, tools do not currently follow this 
-specification (no support for parsing version specifiers, input is case 
-sensitive, ...).
+The ISA subset naming conventions and canonical order are described in
+Chapter `ISA Extension Naming Conventions` of the RISC-V user-level ISA
+specification. However, tools do not currently follow this specification
+(no support for parsing version specifiers, input is case sensitive, ...).
+
+The rule of ISA string become more complicated, due to extensions implication
+rule and more extensions added into RISC-V, the canonical order is non-obvious
+to human, so tools can accept the ISA string in non-canonical order for reduce
+the burden of remembering the canonical order.
+
+Detail rule for ISA string:
+
+1. First letter must be `i`, `e` or `g`.
+2. Single-letter may be non-canonical order.
+3. Multi-letter may be non-canonical order.
+4. Multi-letter must be separated by underscore.
+5. Version separator(`p`) has higher priority than `p` extension.
+
+Example:
+```
+rv32ima_zicsr        # Valid ISA string.
+rv32i_zicsr_m        # Valid ISA string.
+rv32imac             # Valid ISA string.
+rv32mai              # Invalid ISA string, first letter must be `i`, `e` or `g`.
+rv32i_zicsrzifence   # Invalid ISA string, because it will interpreted as rv32
```

Comment:
Is "invalid" the right description here? Isn't it a valid string, but the user probably didn't mean to specify an extension named "zicsrzifence"

Reply:
Yeah, that should be also valid but will interpreted as zicsrzifence, updated :)